packages/nx-plugin/src/generators/init/generator.ts

```diff
@@ -11,41 +11,52 @@ import {
   updateNxJson,
 } from '@nx/devkit';
 import {
-  cpuCliVersion,
-  cpuModelVersion,
-  cpuNxPluginVersion,
-  cpuUtilsVersion,
+  cpCliVersion,
+  cpModelVersion,
+  cpNxPluginVersion,
+  cpUtilsVersion,
 } from '../../utils/versions';
 import { InitGeneratorSchema } from './schema';
 
+const nxPluginPackageName = '@code-pushup/nx-plugin';
+type PkgJson = Record<string, Record<string, string>>;
 function checkDependenciesInstalled(host: Tree) {
-  const packageJson = readJson(host, 'package.json');
+  const packageJson = readJson<PkgJson>(host, 'package.json');
   const devDependencies: Record<string, string> = {};
   const dependencies = {};
-  packageJson.dependencies = packageJson.dependencies || {};
-  packageJson.devDependencies = packageJson.devDependencies || {};
+  packageJson.dependencies = packageJson.dependencies ?? {};
+  packageJson.devDependencies = packageJson.devDependencies ?? {};
 
   // base deps
-  devDependencies['@code-pushup/nx-plugin'] = cpuNxPluginVersion;
-  devDependencies['@code-pushup/models'] = cpuModelVersion;
-  devDependencies['@code-pushup/utils'] = cpuUtilsVersion;
-  devDependencies['@code-pushup/cli'] = cpuCliVersion;
+  devDependencies[nxPluginPackageName] = cpNxPluginVersion;
+  devDependencies['@code-pushup/models'] = cpModelVersion;
+  devDependencies['@code-pushup/utils'] = cpUtilsVersion;
+  devDependencies['@code-pushup/cli'] = cpCliVersion;
 
   return addDependenciesToPackageJson(host, dependencies, devDependencies);
 }
 
 function moveToDevDependencies(tree: Tree) {
-  updateJson(tree, 'package.json', packageJson => {
-    packageJson.dependencies = packageJson.dependencies || {};
-    packageJson.devDependencies = packageJson.devDependencies || {};
+  updateJson(tree, 'package.json', (packageJson: PkgJson) => {
+    const newPackageJson: PkgJson = {
+      dependencies: {},
+      devDependencies: {},
+      ...packageJson,
+    };
 
-    if (packageJson.dependencies['@code-pushup/nx-plugin']) {
-      packageJson.devDependencies['@code-pushup/nx-plugin'] =
-        packageJson.dependencies['@code-pushup/nx-plugin'];
-      delete packageJson.dependencies['@code-pushup/nx-plugin'];
+    if (newPackageJson.dependencies?.[nxPluginPackageName] !== undefined) {
+      const { [nxPluginPackageName]: version, ...dependencies } =
+        newPackageJson.dependencies as { [nxPluginPackageName]: string };
```

Comment:
Type-cast is not needed and also not true (object doesn't only contain `{ '@code-pushup/nx-plugin': string }`).

```suggestion
        newPackageJson.dependencies;
```